cpp/src/arrow/filesystem/util_internal.cc

```diff
@@ -260,6 +263,42 @@ Result<FileInfoVector> GlobFiles(const std::shared_ptr<FileSystem>& filesystem,
   return out;
 }
 
+Result<std::string> CalculateSSECKeyMD5(const std::string& base64_encoded_key,
+    int expect_input_key_size) {
+  if (base64_encoded_key.size() < 2) {
+    return Status::Invalid("At least 2 bytes needed for the base64 encoded string");
+  }
+  // Check if the string contains only valid Base64 characters
+  for (char c : base64_encoded_key) {
+    if (!std::isalnum(c) && c != '+' && c != '/' && c != '=') {
+      return Status::Invalid("Invalid character found in the base64 encoded string");
+    }
+  }
+
+  // Decode the Base64-encoded key to get the raw binary key
+  Aws::Utils::ByteBuffer rawKey =
+      Aws::Utils::HashingUtils::Base64Decode(base64_encoded_key);
```

Comment:
Can we use `arrow::util::base64_decode()` instead?

Reply:
ok, I would refine.